### PR TITLE
DOC Eager loading

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/8426

### CODE REVIEW

File: en/02_Developer_Guides/00_Model/02_Relations.md [hunk 1]
```diff
@@ -570,6 +570,100 @@ the best way to think about it is that the object where the relationship will be
...
+
+# Retrieve all the players for the teams in a single query:
+SELECT * FROM Player WHERE TeamID IN (1, 2, 3, ...)
+```
+
+Eager load multiple relationships by passing additional arguments to the ->eagerLoad() method:
+
+```php
+$teams = Team::get()->eagerLoad('Players', 'Fans');
+```
```
Comment (sabina-talipova) [inline en/02_Developer_Guides/00_Model/02_Relations.md:663]: Could we provide a little bit more details about `multiple relationships` and  `nested relationships`. 
Probably, some code example to show difference. 
And also could we clarify how many `additional arguments` might be passed, for example, like below _"up to **three** levels deep"_, so it means you can request something `Team::get()->eagerLoad('Players', 'Fans', 'Games')` or `Team::get()->eagerLoad('Players.Games', 'Fans')`

**Review by sabina-talipova — CHANGES_REQUESTED** [review]
Looks nice. Probably, it needs to provide a little bit more information how to use `eagerLoad` with some complex relationships.

File: en/02_Developer_Guides/00_Model/02_Relations.md (hunk 1 above)
Comment (emteknetnz) [reply en/02_Developer_Guides/00_Model/02_Relations.md:663]: Updated

File: en/02_Developer_Guides/00_Model/02_Relations.md [hunk 2]
```diff
@@ -570,6 +570,100 @@ the best way to think about it is that the object where the relationship will be
...
+
+```sql
+# Retrieve all the teams
+SELECT * FROM Team
+
+# Retrieve all the players for the teams in a single query:
+SELECT * FROM Player WHERE TeamID IN (1, 2, 3, ...)
+```
+
+Eager load multiple relationships by passing additional arguments separated by a comma to the `eagerLoad()` method. In this example `Players` and `Fans` are `has_many` relationships on the `Team` model:
```
Comment (sabina-talipova) [inline en/02_Developer_Guides/00_Model/02_Relations.md]:
```suggestion
Suppose we have the following architecture and relationships between classes:
```php
class Team extends DataObject
{
    private static $has_many = [
        'Players' => Player::class,
        'Fans' => Fan::class,
    ];
}

class Player extends DataObject
{
    private static $has_one = [
        'Team' => Team::class,
    ];
    
    private static $many_many = [
        'Games' => Game::class,
    ];
}

class Game extends DataObject
{
    private static $has_many = [
         'Players' => Player::class,
        'Officials' => Official::class,
        'Sponsors' => Sponsor::class,
    ];
}
In this example, to get all Players and Fans, you can use eager loading multiple relationships by passing additional arguments separated by a comma to the `eagerLoad()` method:
```

File: en/02_Developer_Guides/00_Model/02_Relations.md [hunk 3]
```diff
@@ -570,6 +570,100 @@ the best way to think about it is that the object where the relationship will be
...
+
+```php
+$teams = Team::get()->eagerLoad('Players', 'Fans');
+```
+
+Eager load nested relationships up to **three** levels deep using the dot syntax. Three levels deep means there will be 2 dots in the relationship. In this example `Games` is a `has_many` relationship on the `Player` model, and `Officials` is a `has_many` relationship on the `Game` model:
+
+```php
+$teams = Team::get()->eagerLoad('Players.Games.Officials')
+
```
Comment (sabina-talipova) [inline en/02_Developer_Guides/00_Model/02_Relations.md]:
```suggestion
Perhaps, you may need to get all the Officials that are related to this Game. In this example, you can use another feature provided by the `eagerLoad` method. Eager load nested relationships up to **three** levels deep using the dot syntax:

```php
$teams = Team::get()->eagerLoad('Players.Games.Officials');

You can even get the results for multiple nested relations by passing them with a comma:

```php
$teams = Team::get()->eagerLoad(
          'Players.Games.Officials', 
          'Players.Games.Sponsors', 
);
```

File: en/02_Developer_Guides/00_Model/02_Relations.md [hunk 4]
```diff
@@ -570,6 +570,100 @@ the best way to think about it is that the object where the relationship will be
...
+    foreach ($team->Players() as $player) {
+        foreach ($player->Games() as $game) {
+            foreach ($game->Officials() as $official) {
+                // Everything will have been eager loaded at this point
+                echo $official->FirstName;
+            }
+        }
+    }
+}
+```
```
Comment (sabina-talipova) [inline en/02_Developer_Guides/00_Model/02_Relations.md:684]:
```suggestion
After receiving the result, you can work with it like with a regular associative array:
```php
foreach ($teams as $team) {
    foreach ($team->Players() as $player) {
        foreach ($player->Games() as $game) {
            foreach ($game->Officials() as $official) {
                // Everything will have been eager loaded at this point
                echo $official->FirstName;
            }
        }
    }
}
```
```

**Review by sabina-talipova — CHANGES_REQUESTED** [review]
Just small suggestions how we can improve and make the examples more sense.

File: en/02_Developer_Guides/00_Model/02_Relations.md (hunk 2 above)
Comment (emteknetnz) [reply en/02_Developer_Guides/00_Model/02_Relations.md]: Updated

**Review by sabina-talipova — APPROVED** [review]
I like it! Nice work!

**Pull request merged**